python/dolfinx/fem/linearproblem.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+import typing
+
+from petsc4py import PETSc
+from ufl import form
+from dolfinx import fem
+
+
+class LinearProblem():
+    """
+    Class for solving a linear variational problem of the form a(u,v)=L(v)
+    using PETSc as a linear algebra backend.
+    """
+
+    def __init__(self, a: form.Form, L: form.Form, bcs: typing.List[fem.DirichletBC] = [],
+                 petsc_options={}, form_compiler_parameters={}, jit_parameters={}):
```

Comment:
Use `foo_options=` for all?

Reply:
`form_compiler_parameters` and `jit_parameters` are standard names used in Form() and jit call.

Reply:
'options' sound better - I think we should change it elsewhere. Who uses 'compiler parameters' rather than 'compiler options'?

Reply:
But this would be a separate PR with overlap in ffcx.

Reply:
Created separate issue #1290